keps/sig-api-machinery/2608-crd-field-level-versioning/README.md

```diff
@@ -0,0 +1,816 @@
+  - [Dependencies](#dependencies)
+  - [Scalability](#scalability)
+  - [Troubleshooting](#troubleshooting)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+## Release Signoff Checklist
+
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [x] (R) Enhancement issue in release milestone, which links to KEP dir in [kubernetes/enhancements] (not the initial KEP PR)
+- [ ] (R) KEP approvers have approved the KEP status as `implementable`
+- [ ] (R) Design details are appropriately documented
+- [ ] (R) Test plan is in place, giving consideration to SIG Architecture and SIG Testing input (including test refactors)
+  - [ ] e2e Tests for all Beta API Operations (endpoints)
+  - [ ] (R) Ensure GA e2e tests for meet requirements for [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+This KEP proposes adding support for field-level versioning for Custom Resources (CRs).
+When a new field is added to an existing, built-in API type, it is typically gated by
+the addition of a feature gate in [`kube_features.go`](https://github.com/kubernetes/kubernetes/blob/master/pkg/features/kube_features.go).
+This gate effectively helps control the lifecycle of the feature that is implemented as
+a result of adding this additional field. Currently it is not possible to version CRs in
+a similar manner. This KEP aims to provide a mechanism that allows feature-gate-like
+functionality for Custom Resources as well.
+
+## Motivation
+
+Currently, it is not possible to have fields of a CR exist in different stages such
+as alpha, beta, stable or deprecated, the CR itself can be in a version that is defined
+by its Custom Resource Definition (CRD). If a field of a CR were to be marked as deprecated,
```

Comment:
I think it is possible to do this, by having different CRD manifests at different stability levels